File: problem_set/best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java

```
// @Title: 买卖股票的最佳时机 (Best Time to Buy and Sell Stock)
// @Author: robert.sunq
// @Date: 2021-06-30 23:38:26
// @Runtime: 3 ms
// @Memory: 51.5 MB

class Solution {
    public int maxProfit(int[] prices) {
        int min  = prices[0];
        int res = 0;
        // pre 到当前天 最大利润
        int pre = 0;
        for(int i = 1;i<prices.length;i++){
            // 当前天的最大利润，是前一天的最大利润，和当前天减去前面最小值的利润中的大者
            pre = Math.max(pre,prices[i] - min);
            min = Math.min(min , prices[i]);
        }
        return pre;
    }
}

File: problem_set/best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp

```
// @Title: 买卖股票的最佳时机 (Best Time to Buy and Sell Stock)
// @Author: robert.sunq
// @Date: 2020-05-12 22:32:24
// @Runtime: 12 ms
// @Memory: 12.7 MB

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int temp[2];
        temp[0] = 0;
        temp[1] =0 ;
        int max = 0 ;
        for(int i=1;i<prices.size();i++){
            if((prices[i]-prices[temp[0]]) > max) {
                max = prices[i]-prices[temp[0]];
                temp[1] = i;
            }
            if(prices[i] < prices[temp[0]]) temp[0] = i;
        }
        return max;
    }
};
